core/server/src/test/java/alluxio/worker/netty/DataServerReadHandlerTest.java

```diff
@@ -0,0 +1,235 @@
+    long checksum = 0;
+    File file = mTestFolder.newFile();
+    long pos = 0;
+    if (length > 0) {
+      FileOutputStream fileOutputStream = new FileOutputStream(file);
+      while (length > 0) {
+        byte[] buffer = new byte[(int) Math.min(length, Constants.MB)];
+        mRandom.nextBytes(buffer);
+        for (int i = 0; i < buffer.length; i++) {
+          if (pos >= start && pos <= end) {
+            checksum += BufferUtils.byteToInt(buffer[i]);
+          }
+          pos++;
+        }
+        fileOutputStream.write(buffer);
+        length -= buffer.length;
+      }
+      fileOutputStream.close();
+    }
+
+    mFile = file.getPath();
+    mockReader(start);
+    return checksum;
+  }
+
+  /**
+   * Checks all the read responses.
+   */
+  protected void checkAllReadResponses(EmbeddedChannel channel, long checksumExpected) {
+    boolean eof = false;
+    long checksumActual = 0;
+    while (!eof) {
+      Object readResponse = waitForOneResponse(channel);
+      if (readResponse == null) {
+        Assert.fail();
+        break;
+      }
+      DataBuffer buffer = checkReadResponse(readResponse, Protocol.Status.Code.OK);
+      eof = buffer == null;
+      if (buffer != null) {
+        if (buffer instanceof DataNettyBufferV2) {
+          ByteBuf buf = (ByteBuf) buffer.getNettyOutput();
+          while (buf.readableBytes() > 0) {
+            checksumActual += BufferUtils.byteToInt(buf.readByte());
+          }
+          buf.release();
+        } else {
+          Assert.assertTrue(buffer instanceof DataFileChannel);
+          ByteBuffer buf = buffer.getReadOnlyByteBuffer();
+          byte[] array = new byte[buf.remaining()];
+          buf.get(array);
+          for (int i = 0; i < array.length; i++) {
+            checksumActual += BufferUtils.byteToInt(array[i]);
+          }
+        }
+      }
+    }
+    Assert.assertEquals(checksumExpected, checksumActual);
+    Assert.assertTrue(eof);
+  }
+
+  /**
+   * Checks the read response message given the expected error code.
+   *
+   * @param readResponse the read response
+   * @param codeExpected the expected error code
+   * @return the data buffer extracted from the read response
+   */
+  protected DataBuffer checkReadResponse(Object readResponse, Protocol.Status.Code codeExpected) {
+    Assert.assertTrue(readResponse instanceof RPCProtoMessage);
+
+    Object response = ((RPCProtoMessage) readResponse).getMessage();
+    Assert.assertTrue(response instanceof Protocol.Response);
+    Assert.assertEquals(codeExpected, ((Protocol.Response) response).getStatus().getCode());
+    return ((RPCProtoMessage) readResponse).getPayloadDataBuffer();
+  }
+
+  /**
+   * Waits for one read response messsage.
+   *
+   * @return the read response
+   */
+  protected Object waitForOneResponse(final EmbeddedChannel channel) {
+    return CommonUtils.waitFor("Response from the channel", new Function<Void, Object>() {
```

Comment:
NIT: I think this does not need to start with capital letter. It will follow "Timed out waiting for "